pkg/ingester/index/multi_test.go

```diff
@@ -0,0 +1,179 @@
+package index
+
+import (
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/astmapper"
+	"github.com/grafana/loki/pkg/storage/config"
+	"github.com/grafana/loki/pkg/storage/stores/tsdb/index"
+)
+
+func MustParseDayTime(s string) config.DayTime {
+	t, err := time.Parse("2006-01-02", s)
+	if err != nil {
+		panic(err)
+	}
+	return config.DayTime{Time: model.TimeFromUnix(t.Unix())}
+}
+
+var testPeriodConfigs = []config.PeriodConfig{
+	{
+		From:      MustParseDayTime("2020-01-01"),
+		IndexType: config.StorageTypeBigTable,
+	},
+	{
+		From:      MustParseDayTime("2021-01-01"),
+		IndexType: config.TSDBType,
+	},
+	{
+		From:      MustParseDayTime("2022-01-01"),
+		IndexType: config.BoltDBShipperType,
+	},
+	{
+		From:      MustParseDayTime("2023-01-01"),
+		IndexType: config.TSDBType,
+	},
+}
+
+// Only run the specific shard factor validation logic if a period config using
+// tsdb exists
+func TestIgnoresInvalidShardFactorWhenTSDBNotPresent(t *testing.T) {
+	factor := uint32(6)
+	_, err := NewMultiInvertedIndex(
+		[]config.PeriodConfig{
+			{
+				From:      MustParseDayTime("2020-01-01"),
+				IndexType: config.StorageTypeBigTable,
+			},
+		},
+		factor,
+	)
+	require.Nil(t, err)
```

Comment:
nitpicking:
I noticed we use `require.Nil(t, err)` frequently, but `require.NoError(t, err)` looks better I think.  [require.NotNil](https://github.com/stretchr/testify/blob/083ff1c0449867d0d8d456483ee5fab8e0c0e1e6/assert/assertions.go#L505) implementation vs [require.NoError implementation](https://github.com/stretchr/testify/blob/083ff1c0449867d0d8d456483ee5fab8e0c0e1e6/assert/assertions.go#L1334)
```suggestion
	require.NoError(t, err)
```

Reply:
I prefer `require.Nil`'s error message when it doesn't pass (helpful debugging regressions/etc), so I usually use that.